wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -85,36 +98,36 @@ public WorkspaceRuntimes(MachineManager machineManager, EventService eventServic
         this.eventService = eventService;
         this.descriptors = new HashMap<>();
         this.startQueues = new HashMap<>();
-        this.rwLock = new ReentrantReadWriteLock();
+        this.addMachineEventSubscriber = new AddMachineEventSubscriber();
+        this.removeMachineEventSubscriber = new RemoveMachineEventSubscriber();
     }
 
     /**
      * Returns the runtime descriptor describing currently starting/running/stopping
      * workspace runtime.
      *
      * <p>Note that the {@link RuntimeDescriptor#getRuntime()} method
-     * returns {@link Optional} which describes just a snapshot copy of
-     * a real {@code WorkspaceRuntime} object, which means that any
-     * runtime copy modifications won't affect the real object and also
-     * it means that copy won't be affected with modifications applied
+     * returns a copy of a real {@code WorkspaceRuntime} object,
+     * which means that any runtime copy modifications won't affect the
+     * real object and also it means that copy won't be affected with modifications applied
      * to the real runtime workspace object state.
      *
      * @param workspaceId
      *         the id of the workspace to get its runtime
      * @return descriptor which describes current state of the workspace runtime
      * @throws NotFoundException
-     *         when workspace with given {@code workspaceId} doesn't have runtime
+     *         when workspace with given {@code workspaceId} is not running
      */
     public RuntimeDescriptor get(String workspaceId) throws NotFoundException {
-        rwLock.readLock().lock();
+        acquireReadLock(workspaceId);
```

Comment:
Well the idea is good, but i will not change the code as it requires an infrastructure under it.
The design of the solution should be moved to the another place like commons and reused later by another clients. The short alias methods are quiet simple and look good for me right now, so i will not change it, but will keep in mind. Thank you